crates/sui-core/src/mysticeti_adapter.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Arc;
+
+use mysticeti_core::block_handler::BlockHandler;
+use mysticeti_core::minibytes;
+use mysticeti_core::types::{BaseStatement, StatementBlock};
+
+use sui_types::error::{SuiError, SuiResult};
+use tap::prelude::*;
+use tokio::sync::{mpsc, oneshot};
+
+use crate::authority::authority_per_epoch_store::AuthorityPerEpochStore;
+use crate::consensus_adapter::SubmitToConsensus;
+use mysticeti_core::data::Data;
+use sui_types::messages_consensus::ConsensusTransaction;
+use tracing::warn;
+
+#[derive(Clone)]
+pub struct SubmitToMysticeti {
+    sender: mpsc::Sender<(ConsensusTransaction, oneshot::Sender<()>)>,
+}
+
+impl SubmitToMysticeti {
+    pub fn new(
+        sender: mpsc::Sender<(ConsensusTransaction, oneshot::Sender<()>)>,
+    ) -> SubmitToMysticeti {
+        SubmitToMysticeti { sender }
+    }
+}
+
+#[async_trait::async_trait]
+impl SubmitToConsensus for SubmitToMysticeti {
+    async fn submit_to_consensus(
+        &self,
+        transaction: &ConsensusTransaction,
+        _epoch_store: &Arc<AuthorityPerEpochStore>,
+    ) -> SuiResult {
+        let (sender, receiver) = oneshot::channel();
+        self.sender
+            .send((transaction.clone(), sender))
+            .await
+            .tap_err(|e| warn!("Submit transaction failed with {:?}", e))
+            .map_err(|e| SuiError::FailedToSubmitToConsensus(format!("{:?}", e)))?;
+        // Give a little bit backpressure if BlockHandler is not able to keep up.
+        receiver
```

Comment:
I guess the SubmitToMysticeti:: sender channel it self could also work as a backpressure mechanism so maybe we don't _really_ need that here, but not opposed to have that either.

Reply:
right this is not changing anything significantly at the moment